utils/emissions.go

```diff
@@ -0,0 +1,368 @@
+package utils
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ava-labs/avalanchego/api/info"
+	"github.com/chain4travel/magellan/cfg"
+	"github.com/chain4travel/magellan/models"
+)
+
+const (
+	perYear  string = "Per Year"
+	perMonth string = "Per Month"
+	perDay   string = "Per Day"
+)
+
+var countryIDs = []string{"TAIWAN", "UNITED_STATES", "GERMANY", "NETHERLANDS", "BELGIUM", "FINLAND"}
+
+func GetDailyEmissions(startDate time.Time, endDate time.Time, config cfg.EndpointService, rpc string) models.Emissions {
+	dailyEmissions := []models.EmissionsResult{}
+	startDatef := strings.Split(startDate.String(), " ")[0]
+	endDatef := strings.Split(endDate.String(), " ")[0]
+	chainIDs, _ := accessibleChains(config)
+	for _, chain := range chainIDs {
+		intensityFactor, err := CarbonIntensityFactor(chain, startDatef, endDatef, config)
+		if len(intensityFactor) > 0 && err == nil {
+			dailyEmissions = append(dailyEmissions, models.EmissionsResult{
+				Chain: chain,
+				Value: getAvgEmissionsValue(intensityFactor),
+			})
+		}
+	}
+	return models.Emissions{Name: "Daily emissions", Value: dailyEmissions}
+}
+
+func GetNetworkEmissions(startDate time.Time, endDate time.Time, config cfg.EndpointService, rpc string) (*models.Emissions, error) {
+	startDatef := strings.Split(startDate.String(), " ")[0]
+	endDatef := strings.Split(endDate.String(), " ")[0]
+	infoClient := info.NewClient(rpc)
+	networkName, _ := infoClient.GetNetworkName(context.Background())
+	emissionsResult := []*models.EmissionsResult{}
+	emissions := &models.Emissions{
+		Name:  "Network Emissions",
+		Value: emissionsResult,
+	}
+	NetworkInmutable, err := getNetworkNameInmutable(networkName, config)
+	if NetworkInmutable != "" {
+		emissionsResult, err = network(NetworkInmutable, startDatef, endDatef, config)
+		if err != nil || emissionsResult == nil {
+			return emissions, err
+		}
+		err = getEmissionsResults(emissions, emissionsResult, startDate, endDate)
+	}
+	return emissions, err
+}
+
+func GetCountryEmissions(startDate time.Time, endDate time.Time, config cfg.EndpointService, rpc string) (models.Emissions, error) {
+	var (
+		err         error
```

Comment:
This could be omitted i think. And replace `countryInfo, err = country(countryID, startDatef, endDatef, config)` with `countryInfo, err := country(countryID, startDatef, endDatef, config)`